### PR TITLE
vendor: seelog constrain to version 2.6

### PR DESCRIPTION
**Pull request by adnxn**
### Summary

dep best practices, constrain seelog to version 2.6 because we were tied to a githash. Updated dependencies using [dep v0.4.1](https://github.com/golang/dep/releases/tag/v0.4.1):
```
% dep version
dep:
 version     : v0.4.1
 build date  : 2018-01-24
 git hash    : 37d9ea0a
 go version  : go1.9.1
 go compiler : gc
 platform    : linux/386
```
### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

This contribution is under the terms of the Apache 2.0 License:  yes

### CODE REVIEW

File: agent/Gopkg.toml [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 [[constraint]]
   name = "github.com/cihub/seelog"
-  revision ="0fa7e41d2b9600ca6b7e1592478124e4f800f724"
+  version ="2.6"
```
Comment (aaithal) [inline agent/Gopkg.toml]: was this a manual edit? Curious to know the best practices in dep land.

Comment (adnxn) [reply agent/Gopkg.toml]: yea this was a manual edit, we always want to constrain to fixed release version where possible. The generated "Gopkg.lock' file will map it to the appropriate [hash here](https://github.com/aws/amazon-ecs-agent/pull/1232/files#diff-0f11b02fdde503182f627b67a8f1876bR66).

Comment (aaithal) [reply agent/Gopkg.toml]: thanks for that explanation. probably a good time to update my `dep` from `dev` version to something else as well.

Comment (adnxn) [reply agent/Gopkg.toml]: updated PR summary with dep version info, we should include that information in PRs for dep generated changes to the repo.

**Pull request merged**